lib/domain/services/message_sync_service.dart

```diff
@@ -305,4 +311,107 @@ class MessageSyncService {
       _logger.warning('Failed to set last sync time for group $groupId', e);
     }
   }
+
+  /// Filters invites to find new ones that should trigger notifications.
+  ///
+  /// Excludes invites that have already been notified about.
+  ///
+  /// Returns an empty list for invalid inputs.
+  static Future<List<Welcome>> filterNewInvites({
+    required String activePubkey,
+    required List<Welcome> welcomes,
+  }) async {
+    if (activePubkey.isEmpty) {
+      _logger.warning('Empty activePubkey provided to filterNewInvites');
+      return [];
+    }
+
+    try {
+      final prefs = await _preferences;
+      final notifiedIds = prefs.getStringList('bg_sync_notified_invites_$activePubkey') ?? [];
+      final notifiedSet = notifiedIds.toSet();
+
+      final newWelcomes = welcomes.where((welcome) => !notifiedSet.contains(welcome.id)).toList();
```

Comment:
This seems inefficient, won't the `notifiedIds` list grow boundlessly??

Maybe we can do some timestamp-based filtering first?